plugins/c8y_configuration_plugin/src/download.rs

```diff
@@ -315,28 +313,22 @@ impl ConfigDownloadManager {
 
         if file_path.exists() {
             // Confirm that the file has write access before any http request attempt
-            config_download_request.has_write_access()?;
+            has_write_access(file_path.as_path())?;
         } else if let Some(file_parent) = file_path.parent() {
             if !file_parent.exists() {
                 fs::create_dir_all(file_parent)?;
```

Comment:
There is no reasons to change this for that PR. However, I think we need to reconsider this creation of all the missing parents. The issue with that is that is unlikely the all directories up to the root dir have to be owned by the same user. The tests were failing because trying to create `/var/tedge/file-transfer/child-device/config_update` starting from `/var`. Neither `root ` nor the `tedge` user can do that correctly.